resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/backend/minikube/Minikube.scala

```diff
@@ -38,12 +38,50 @@ private[spark] object Minikube extends Logging {
 
   def getMinikubeStatus: MinikubeStatus.Value = {
     val statusString = executeMinikube("status")
+    logInfo(s"Minikube status command output:\n$statusString")
+    // up to minikube version 0.30 use this to check for minikube status
+    val oldMinikube = statusString
       .filter(line => line.contains("minikubeVM: ") || line.contains("minikube:"))
-      .head
-      .replaceFirst("minikubeVM: ", "")
-      .replaceFirst("minikube: ", "")
-    MinikubeStatus.unapply(statusString)
+
+    if (oldMinikube.isEmpty) {
```

Comment:
This code is a bit hard to read due to the multiple levels of nesting with `if-else` blocks. Can you put the code for the newer versions of Minikube into a separate method?

Reply:
sure